src/__tests__/Relay.test.ts

```diff
@@ -0,0 +1,82 @@
+import Relay from '../Relay';
+
+describe('Relay', () => {
+  it('sends a message to a connection', () => {
+    const relay = new Relay();
+    const callback = jest.fn();
+    relay.addConnection('Bedford Ave', callback);
+
+    relay.send('The L train is delayed.', {
+      to: 'Bedford Ave',
+    });
+
+    expect(callback).toBeCalledWith({ 
+      message: 'The L train is delayed.', 
+      to: undefined,
+    });
+  });
+
+  it('sends the next destination to the connection', () => {
+    const relay = new Relay();
+    const callback = jest.fn();
+    relay.addConnection('Bedford Ave', callback);
+
+    relay.send('The L train is delayed.', {
+      to: 'Bedford Ave:Lorimer St',
+    });
+
+    expect(callback).toBeCalledWith({ 
+      message: 'The L train is delayed.', 
+      to: 'Lorimer St'
+    });
+  });
+
+  it('can remove a connection', () => {
+    const relay = new Relay();
+    const callback = jest.fn();
+    relay.addConnection('Bedford Ave', callback);
+
+    relay.send('The L train is delayed.', {
+      to: 'Bedford Ave',
+    });
+
+    expect(callback).toBeCalledWith({ 
+      message: 'The L train is delayed.', 
+      to: undefined,
+    });
+
+    relay.removeConnection('Bedford Ave');
+
+    relay.send('The L train is still delayed.', {
+      to: 'Bedford Ave',
+    });
+
+    expect(callback).not.toBeCalledTimes(2);
```

Comment:
Speaking from painful experience, tests that expect things to happen synchronously can become a real liability down the road, especially when it's something like sending/receiving a message, which you might want to be asynchronous at some point in the future.

A different approach is to use the `resolve` function from `new Promise(resolve => ...)` as the callback, and return a `Promise` from the test, so the test will pass if/when the `resolve` function is called, or time out after 5000ms if it isn't called. It may also be tempting to use `async`/`await`, but that doesn't work so well when there are callback functions involved.